rep-2014.rst

```diff
@@ -0,0 +1,238 @@
+
+In another example, a robotic system that can perform a task in 1 second with a power consumption of `2W` is twice as fast (*latency*) as another robotic system that can perform the same task in 2 seconds with a power consumption of `0.5W`. However, the second robotic system is twice as efficient as the first one. In this case, the solution that requires less power would be the best option from an energy efficiency perspective (with a higher *performance-per-watt*). Similarly, a robotic system that has a high bandwidth but consumes a lot of energy might not be the best option for a mobile robot that must operate for a long time on a battery.
+
+Therefore, it is important to consider different of these metrics when benchmarking a robotic system. The metrics presented in this REP are intended to be used as a guideline, and should be adapted to the specific needs of a robot.
+
+
+Methodology for benchmarking performance in ROS 2
+=================================================
+
+In this REP, we **recommend adopting a grey-box and non-functional benchmarking approach** to measure performance and allow to evaluate ROS 2 individual nodes as well as complete computational graphs. To realize it in an architecture-neutral, representative, and reproducible manner, we also recommend using the Linux Tracing Toolkit next generation (`LTTng <https://lttng.org/>`_) through the `ros-tracing` project, which leverages probes already inserted in the ROS 2 core layers and tools to facilitate benchmarking ROS 2 abstractions.
+
+The following diagram shows the proposed methodology for benchmarking performance in ROS 2 which consists of 3 steps:
+
+::
+
+                                                                
+                                                +--------------+
+                    +----------------+  rebuild |              |
+                    |                +---------->              |
+  start  +----------> 1. trace graph |          | 2. benchmark +----------> 3. report
+                    |                |          |              |
+                    +----+------^--^-+          |              |
+                         |      |  |            +-------+------+
+                         |      |  |                    |
+                         +------+  |                    |
+                           LTTng   +--------------------+
+                                       re-instrument
+
+
+1. instrument both the target ROS 2 abstraction/application using `LTTng <https://lttng.org/>`_. Refer to `ros2_tracing <https://github.com/ros2/ros2_tracing>`_ for tools, documentation and ROS 2 core layers tracepoints;
+2. trace and benchmark the ROS 2 application;
```

Comment:
I think this would be a good place for technical details. Current C/C++ benchmarks in ROS 2 use Google Benchmark, and are skipped unless `AMENT_RUN_PERFORMANCE_TESTS` is explicitly set.